syntax = "proto3";

package list_trade_solicitations;

service ListTradeSolicitations {
  rpc makeListTradeSolicitationsFactory (Request) returns (Response);
}

message Request {
  int32 id = 1;
}

message Response {
  int32 statusCode = 1;
  repeated TradeList tradeList = 2;
  optional string message = 3;
}

message TradeList {
  int32 received_user_id = 1;
  int32 sender_user_id = 2;
  int32 want_pokemon_id = 3;
  int32 give_pokemon_id = 4;
  string status = 5;
  string want_pokemon_name = 6;
  string want_pokemon_image = 7;
  string give_pokemon_name = 8;
  string give_pokemon_image = 9;
  string received_user_name = 10;
  string sender_user_name = 11;
}